python/ray/includes/object_ref.pxi

```diff
@@ -166,8 +166,19 @@ cdef class ClientObjectRef(ObjectRef):
             # call_release in this case, since the client should have already
             # disconnected at this point.
             return
-        if client.ray.is_connected() and not self.data.IsNil():
-            client.ray.call_release(self.id)
+        if client.ray.is_connected():
+            try:
+                self._wait_for_id(timeout=10)
+            # cython would suppress this exception as well, but it tries to
+            # print out the exception which may crash. Log a simpler message
+            # instead.
+            except Exception:
+                logger.info(
+                    "Exception in ObjectRef is ignored in destructor. "
+                    "To raise the exception, use the object reference before "
+                    "its destructor runs.")
```

Comment:
Exceptions cannot be raised from destructors and trying to print them can crash too. If the exception is a concern to users, it can be acted upon by calling `ray.get()` on the object ref before the object ref is destroyed, to raise the exception. Is there a better way to reword this?

Reply:
Let me make take a stab at fixing it!